kuzzle-sdk-java/src/main/java/io/kuzzle/sdk/Kuzzle.java

```diff
@@ -0,0 +1,284 @@
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import io.kuzzle.sdk.CoreClasses.Responses.*;
+
+import static io.kuzzle.sdk.Helpers.Default.notNull;
+
+public class Kuzzle {
+
+    protected EventListener tokenExpiredEvent;
+    protected EventListener<Response> unhandledResponseEvent;
+
+    protected final AbstractProtocol networkProtocol;
+
+    public final String version;
+    public final String instanceId;
+
+    /**
+     * Authentication token
+     */
+    protected AtomicReference<String> authenticationToken;
+
+
+    /**
+     * The maximum amount of elements that the queue can contains.
+     * If set to -1, the size is unlimited.
+     */
+    protected AtomicInteger maxQueueSize;
+
+
+    /**
+     * The minimum duration of a Token before being automatically refreshed.
+     * If set to -1 the SDK does not refresh the token automatically.
+     */
+    protected AtomicInteger minTokenDuration;
+
+
+    /**
+     * The minimum duration of a Token after refresh.
+     * If set to -1 the SDK does not refresh the token automatically.
+     */
+    protected AtomicInteger refreshedTokenDuration;
+
+    /**
+     * The maximum delay between two requests to be replayed
+     */
+    protected AtomicInteger maxRequestDelay;
+
+    protected ConcurrentHashMap<String, Task<Response>>
+                    requests = new ConcurrentHashMap<>();
+
+    /** Initialize a new instance of Kuzzle
+     * @param networkProtocol   The network protocol
+     * @throws IllegalArgumentException
+     */
+    public Kuzzle(AbstractProtocol networkProtocol)
+            throws IllegalArgumentException {
+        this(networkProtocol, new KuzzleOptions());
+    }
+
+    /** Initialize a new instance of Kuzzle
+     * @param networkProtocol   The network protocol
+     * @param options           Kuzzle options
+     * @throws IllegalArgumentException
+     */
+    public Kuzzle(
+            AbstractProtocol networkProtocol,
+            final KuzzleOptions options
+    ) throws IllegalArgumentException {
+
+        if (networkProtocol == null) {
+            throw new IllegalArgumentException("networkProtocol can't be null");
+        }
+
+        KuzzleOptions kOptions = options != null
+                ? options
+                : new KuzzleOptions();
+
+        this.networkProtocol = networkProtocol;
+        this.networkProtocol.registerResponseEvent(this::onResponseReceived);
+        this.networkProtocol.registerStateChangeEvent(this::onStateChanged);
+
+        this.maxQueueSize = new AtomicInteger(kOptions.getMaxQueueSize());
+        this.minTokenDuration = new AtomicInteger(kOptions.getMinTokenDuration());
+        this.refreshedTokenDuration = new AtomicInteger(kOptions.getRefreshedTokenDuration());
+        this.maxRequestDelay = new AtomicInteger(kOptions.getMaxRequestDelay());
+
+        this.version = "3.0.0";
```

Comment:
This should be dynamically inferred from the `build.graddle` file

Reply:
Yeah I know, but for know I can't manage to do it, just because I'm strugguling with gradle and it just doesn't work, so it would be done but for now, I can't do it.

Feel free to help out me if you want.